lib/mongoid/contextual/aggregable/mongo.rb

```diff
@@ -124,7 +124,7 @@ def sum(field = nil)
         def pipeline(field)
           db_field = "$#{database_field_name(field)}"
           pipeline = []
-          pipeline << { "$match" => criteria.nin(field => nil).selector }
+          pipeline << { "$match" => { field => criteria.selector[field.to_s] } }
```

Comment:
In your case you are guaranteed the field is there, but in this case you are not:

```ruby
account.campaigns.where(name: '...').max(:number)
```

So we just need to make sure that when the field does not exist, the calculations don't fail.